docs/spec_draft.md

```diff
@@ -1443,6 +1444,138 @@ specification. Numeric precision is implementation-defined.
+   if
+   * `jd = floor(((padding[d] + id) + jd * (window_dimensions[d] - window_strides[d])) /  window_dimensions[d])`.
+   * `selected_indices[j0, j1, ..., jR-1] = (i0, i1, ..., iR'-1)`.
+ * `result[i0, i1, ..., iR'-1] = init_value`, otherwise.
+
+The evaluation order of the `scatter` function is arbitrary and may be
+non-deterministic. Therefore, the scatter function should not be overly
+sensitive to reassociation. See the discussion about associativity in the
+context of [stablehlo.reduce](#stablehloreduce) for more details.
+
+### Inputs
+
+| Name                | Type                                      | Constraints      |
+|---------------------|-------------------------------------------|------------------|
+| `operand`           | tensor of any supported type              | (C1-C10)         |
+| `source`            | tensor of any supported type              | (C1-C3)          |
+| `init_value`        | scalar of any supported type              | (C4), (C9)       |
+| `window_dimensions` | 1-dimensional tensor constant type `si64` | (C1), (C3), (C5) |
+| `window_strides`    | 1-dimensional tensor constant type `si64` | (C6)             |
+| `padding`           | 2-dimensional tensor constant type `si64` | (C3), (C7)       |
+| `select`            | `function`                                | (C8)             |
+| `scatter`           | `function`                                | (C4), (C9)       |
+
+### Outputs
+
+| Name     | Type                         | Constraints |
+|----------|------------------------------|-------------|
+| `result` | tensor of any supported type | (C9-C10)    |
+
+### Constraints
+
+  * (C1) rank(`source`) $=$ rank(`operand`) $=$ size(`window_dimensions`).
+
+  * (C2) `operand` and `source` have the same element type ignoring floating-point
+    precision.
+
+  * (C3) `dim(source, i) = strided_bound[i]` for any i $\in$ axis(`source`)
+    * `strided_bound[i] = (padded_operand[i] == 0 || window_dimensions[i] > padded_operand[i]) ? 0 : floor((padded_operand[i] - window_dimensions[i]) / window_strides[i]) + 1`
+    * `padded_operand[i] = padding[i][0] + dim(operand, i) + padding[i][1]`
+
+  * (C4) `init_value` value has to form an identity under `scatter`.
+         The behavior of the operation is undefined otherwise.
+
+  * (C5) On `window_dimensions`
+    * size(`window_dimensions`) $=$ rank(`operand`).
+    * `window_dimensions[i]` $\gt 0$ for all i $\in$ [0, size(window_dimensions)).
+
+  * (C6) On `window_strides` (optional)
+    * Default value of 1 for each dimension of `operand`.
+    * size(`window_strides`) $=$ rank(`operand`).
+    * `window_strides[i]` $\gt 0$ for all i $\in$ [0, size(window_strides)).
+
+  * (C7) On `padding`(optional)
+    * Default value of (0,0) for each dimension of `operand`.
+    * size(`padding`) $=$ rank(`operand`).
+    * dim(`padding`, 0) $=$ N and dim(`padding`, 1) = 2.
+
+  * (C8) `select` has type `(T, T) -> i1` such that T is a 0-D tensor with
```

Comment:
Could we maybe simplify this to refer to stablehlo.select which doesn't yet exist since it seems to be defining what the select operation should be doing.

Reply:
My idea is to re-use `stablehlo.reduce_window` and that the `select` function needs to be used as the `reduce_window::body`. But the problem was  `reduce_window::body` constraints with type `(T, T) -> T` are different that that of `select` with type ``(T, T) -> i1`. Rewrote the formulation to fit them in. 

Reusing `reduce_window` can effectively simplify the operation. Let me know if you have follow up feedback.